File: db/batch_update.py
import json
import os

DB_PATH = r'c:\Users\heroy\COL_ENG\db\20260101_COL_ENG_1484_backupjson'

def load_db():
    with open(DB_PATH, 'r', encoding='utf-8') as f:
        return json.load(f)

def save_db(data):
    with open(DB_PATH, 'w', encoding='utf-8') as f:
        json.dump(data, f, ensure_ascii=False, indent=2)

def get_batch(start_idx, size=50):
    data = load_db()
    return data[start_idx : start_idx + size]

def update_chinese_from_file(trans_path):
    with open(trans_path, 'r', encoding='utf-8') as f:
        translations = json.load(f)
    # Convert keys to int if they are strings
    translations = {int(k): v for k, v in translations.items()}
    
    data = load_db()
    for item in data:
        item_id = item.get('id')
        if item_id in translations:
            item['chinese'] = translations[item_id]
    save_db(data)
    print(f"Updated {len(translations)} records.")

if __name__ == "__main__":
    # This script is intended to be used via run_command with python -c or as a module
    pass

File: db/batch_update.js
const fs = require('fs');
const path = require('path');

const DB_PATH = 'c:/Users/heroy/COL_ENG/db/20260101_COL_ENG_1484_backupjson';

function updateBatch(transPath) {
    const translations = JSON.parse(fs.readFileSync(transPath, 'utf8'));
    const data = JSON.parse(fs.readFileSync(DB_PATH, 'utf8'));
    
    let count = 0;
    data.forEach(item => {
        if (translations[item.id]) {
            item.chinese = translations[item.id];
            count++;
        }
    });
    
    fs.writeFileSync(DB_PATH, JSON.stringify(data, null, 2), 'utf8');
    console.log(`Updated ${count} records.`);
}

if (require.main === module) {
    const transPath = process.argv[2];
    if (transPath) {
        updateBatch(transPath);
    } else {
        console.error('Usage: node batch_update.js <path_to_translations_json>');
    }
}
